airbyte-integrations/connectors/source-webflow/source_webflow/source.py

```diff
@@ -0,0 +1,330 @@
+
+    """This is the main class that defines the methods that will be called by Airbyte infrastructure"""
+
+    def get_collection_name_to_id_dict(self, authenticator: str = None, site_id: str = None) -> Mapping[str, str]:
+        """
+        Most of the Webflow APIs require the collection id, but the streams that we are generating use the collection name.
+        This function will return a dictionary containing collection_name: collection_id entries.
+        """
+
+        collection_name_to_id_dict = {}
+
+        collections_stream = Collections(authenticator=authenticator, site_id=site_id)
+        collections_records = collections_stream.read_records(sync_mode="full_refresh")
+
+        # Loop over the list of records and create a dictionary with name as key, and _id as value
+        for collection_obj in collections_records:
+            collection_name_to_id_dict[collection_obj["name"]] = collection_obj["_id"]
+
+        return collection_name_to_id_dict
+
+    @staticmethod
+    def get_authenticator(config):
+        """
+        Verifies that the information for setting the header has been set, and returns a class
+        which overloads that standard authentication to include additional headers that are required by Webflow.
+        """
+        api_key = config.get("api_key", None)
+        accept_version = WEBFLOW_ACCEPT_VERSION
+        if not api_key:
+            raise Exception("Config validation error: 'api_key' is a required property")
+
+        return WebflowTokenAuthenticator(token=api_key, accept_version=accept_version)
+
+    def check_connection(self, logger: logging.Logger, config: Mapping[str, Any]) -> Tuple[bool, any]:
+        """
+        A check to validate that the user-provided config can be used to connect to the underlying API
+
+        :param config:  the user-input config object conforming to the connector's spec.yaml
+        :param logger:  logger object
+        :return Tuple[bool, any]: (True, None) if the input config can be used to connect to the API successfully, (False, error) otherwise.
+        """
+
+        try:
+            # Check that authenticator can be retrieved
+            authenticator = self.get_authenticator(config)
+            site_id = config.get("site_id")
+            collections_stream = Collections(authenticator=authenticator, site_id=site_id)
+            collections_records = collections_stream.read_records(sync_mode="full_refresh")
+            record = next(collections_records)
+            logger.info(f"Successfully connected to Collections stream. Pulled one record: {record}")
+            return True, None
+        except Exception as e:
+            return False, e
+
+    def generate_streams(self, authenticator: WebflowTokenAuthenticator, site_id: str) -> List[Stream]:
+        """ "Generates a list of stream by their names."""
+
+        collection_name_to_id_dict = self.get_collection_name_to_id_dict(authenticator=authenticator, site_id=site_id)
```

Comment:
Would it be possible to also get a unit test for this under the test_streams.py file?